client/app/views/folder.coffee

```diff
@@ -219,121 +218,118 @@ module.exports = class FolderView extends BaseView
+                next()
+
+        _addDirectories = (items, path='/', callback) ->
+            async.eachSeries items, (entry, next) ->
+                # 1rst case : event data type
+                # otherwhise its a recursive call
+                if entry instanceof DataTransferItem
+                    entry = entry.webkitGetAsEntry()
+
+                # Save file
+                if entry.isFile
+                    _saveEntry entry, path, next
+
+                # Save directory
+                # and its subfiles and/or subdirectory
+                else if entry.isDirectory
+                    entry.createReader().readEntries (entries) ->
+                        path = entry.fullPath.substring 1
+                        _addDirectories entries, path, next
+
+            , (args...) ->
+                # Save all files or goto nextDirectory
+                # it depends on recursive or not call
+                callback() if callback?
+
+
+        if (items = event.dataTransfer?.items or event.target?.items)
+            _addDirectories items, null, () =>
+                _success = -> @uploadQueue.addFolderBlobs files, @model
+
+                # Show Upload Errors
+                if errors.length
+                    formattedErrors = errors
+                        .map (name) -> "\"#{name}\""
+                        .join ', '
+                    localeOptions =
+                        files: formattedErrors
+                        smart_count: errors.length
+
+                    new Modal t('chrome error dragdrop title'), \
+                        t('chrome error dragdrop content', localeOptions), \
+                        t('chrome error submit'), null, _success
+
+                # Show files uploaded
+                else
+                    _success()
 
-                new Modal t('chrome error dragdrop title'), \
-                    t('chrome error dragdrop content', localeOptions), \
-                    t('chrome error submit'), null, (confirm) -> processUpload()
-            else
-                processUpload()
-
-
-        # An entry can be a folder or a file.
-        parseEntriesRecursively = (entry, path, done) =>
-            path = path or ""
-            path = "#{path}/" if path.length > 0
-
-            # If it's a file we add it to the file list with a proper
-            # relative path.
-            if entry.isFile
-                entry.file (file) ->
-                    file.relativePath = "#{path}#{file.name}"
-                    files.push file
-                    done()
-                , (error) ->
-                    errors.push entry.name
-                    done()
-
-            # If it's a directory, recursively call the function to reach
-            # the leaves of the file tree.
-            else if entry.isDirectory
-                reader = entry.createReader()
-
-                # .readEntries only return chunks of 100 elements, so it must
```

Comment:
Did the new code handle this special case? If it's still true about ```.readEntries```, the recursive function [_addDirectories](https://github.com/cozy/cozy-files/pull/464/files#diff-d3d3b1e33365cf96c153a5305affcab1R285) here seems to result with a folder of 100 files in this case.